UnityProject/Assets/Scripts/Items/EncryptionKey.cs

```diff
@@ -309,6 +208,19 @@ public void ServerPerformInteraction(MouseDrop interaction)
 			radio.AddEncryptionKey(this);
 		}
 	}
+
+	public bool Interact(InventoryApply interaction)
+	{
+		//insert the headset key if this is used on a headset
+		if (interaction.UsedObject == gameObject
+		    && interaction.TargetObject.GetComponent<Headset>() != null)
+		{
+			UpdateHeadsetKeyMessage.Send(interaction.TargetObject, gameObject);
```

Comment:
Is this the only place where UpdateHeadsetKeyMessage is called?

If it is and if you have time could you refactor this to use `ICheckedInteractable`, otherwise I'll make an issue to refactor it later.